packages/truffle-db/src/workspace/test/index.ts

```diff
@@ -249,274 +276,221 @@ mutation AddCompilation($compilerName: String!, $compilerVersion: String!, $sour
-
-      expect(source).toHaveProperty("contents");
-      const { contents } = source;
-      expect(contents).not.toBeNull();
-    }
-  }
+  });
 });
 
-const getContractConstructors = `
-query contractConstructor($id: ID!) {
-  contractConstructor(id:$id) {
-    createBytecode {
-      bytes
-      sourceMap
-    }
-    contract {
+/*
+ * Contract
+ */
+
+const GetContract = gql`
+query getContract($id:ID!){
+    contract(id:$id) {
       name
-      source {
-        sourcePath
-      }
-    }
-    compilation {
-      compiler {
-        name
+      abi {
+        json
       }
-      contracts {
-        name
+      sourceContract {
         source {
-          sourcePath
+          contents
+        }
+        ast {
+          json
         }
-      }
-      sources {
-        sourcePath
-        contents
-        ast  
       }
     }
-  }
 }`
 
-const AddContractConstructors = `
-mutation addConstructors($bytecodeId: ID!, $compilationId: ID!, $contractId: ID!) {
-    contractConstructorsAdd(input: {
-      contractConstructors:[{
+const AddContracts = gql`
+mutation addContracts($contractName: String, $compilationId: ID!, $bytecodeId:ID!, $abi:String!) {
+  contractsAdd(input: {
+    contracts: [{
+      name: $contractName
+      abi: {
+        json: $abi
+      }
+      compilation: {
+        id: $compilationId
+      }
+      sourceContract: {
+        index: 0
+      }
+      constructor: {
         createBytecode: {
           id: $bytecodeId
         }
-        compilation: {
-          id: $compilationId
+      }
+    }]
+  }) {
+    contracts {
+      id
+      name
+      sourceContract {
+        name
+        source {
+          contents
         }
-        contract: {
-          id: $contractId
+        ast {
+          json
         }
-      }]
-    }) {
-      contractConstructors {
-        id
+      }
+      constructor {
         createBytecode {
           bytes
-          sourceMap
-        }
-        compilation {
-          compiler {
-            name
-          }
-          contracts {
-            name
-          }
-          sources {
-            sourcePath
-          }
-        }
-        contract {
-          name
-          source {
-            sourcePath
-          }
         }
       }
     }
+  }
 }`
 
-it("adds contract constructors", async () => {
-  const workspace = new Workspace();
+describe("Contract", () => {
+  const client = new WorkspaceClient();
 
   const variables = {
-    bytecodeId: soliditySha3(Migrations.bytecode), 
-    compilationId: "0xfd63faebf4e7515b2f48fdf22066869e09df9b2e91f4e0cf88d7d14ec496b515",
-    contractId: soliditySha3(Migrations.contractName, soliditySha3(Migrations.source, Migrations.sourcePath))
+    contractName: Migrations.contractName,
+    compilationId: compilationId,
+    id: contractId,
+    compilerName: Migrations.compiler.name,
+    compilerVersion: Migrations.compiler.version,
+    sourceId: sourceId,
+    bytecodeId: bytecodeId, 
```

Comment:
For `compilationId`, `sourceId`, and `bytecodeId`, let's avoid calculating them for this test, and instead add source and bytecode manually to the before block and read the IDs from the result.